spec/lib/insights/api/common/open_api/serializer_spec.rb

```diff
@@ -0,0 +1,174 @@
+describe Insights::API::Common::OpenApi::Serializer do
+  describe "#as_json" do
+    class TestSourceType < SourceType
```

Comment:
Having run into issues with constant collisions in specs before, I would recommend using anonymous classes here.  See https://github.com/ManageIQ/manageiq/blob/78cecce6e0a3954d3d14f3b270e0c39ab057b9ea/spec/models/mixins/miq_request_mixin.rb#L2-L13 for an example.  You can do `Class.new(SourceType) do ...` if you need to inherit.

Reply:
Haha alright I'll try to go that route, I know when I went to use `TestClass < ActiveRecord` I ran into issues so that was why I wanted to directly inherit from `SourceType` but it seems like what you're suggesting here should work, so I'll fix it, hopefully